extra_data/components.py

```diff
@@ -396,7 +396,9 @@ def get_dask_array(self, key, subtrain_index='pulseId'):
 
             arrays.append(mod_arr)
 
-        return xarray.concat(arrays, pd.Index(modnos, name='module'))
+        # set the fill_value to prevent xarray from changing the dtype to float
+        fill_value = 0 if arrays[0].dtype.kind != 'f' else xarray.core.dtypes.NA
```

Comment:
Looking at the code in xarray, I think the `NA` sentinel always gets turned into `np.nan` for floats, if you want a shorter way to write it.

There are also other dtypes - complex, object, datetime, timedelta - which xarray is prepared to handle, though I don't think our data would ever be in those.